### PR TITLE
[MXNET-771] Fix Flaky Test test_executor.py:test_dot

### LINKED ISSUE #11686: test_executor.test_dot has fixed seed that can mask flakiness
The unit test in title have been using fixed seed to mask flakiness. Suggested action:
1. Evaluate whether the test is flaky without fixed seed. If not, remove seed. Else move to 2
2. If test is flaky, determine whether it's an actual uncaught edge case. If so, fix the operator. Else move to 3
3. If numerical instability is inevitable, adjust tolerance level appropriately.

### PR DESCRIPTION
**Pull request by roywei**
## Description ##
Fix #11686, removed fixed seed in test_executor.py:test_dot

1. Verified this test is flaky and the values for comparison are slightly off
2. Flakiness comes from sym.dot of 2 large tensors. The input tensor size was original set to be a random size in range of (1,500). Reduced 500 to 200 to make it more stable.
3.  The test still fails with 1e-6 tolerance, increased it to 1e-5 (reasonable for fp32)
4. removed helper function `reldiff` (calculating relative difference) with `assert_almost_equal` in `mxnet.test_utils`  for value comparison
5. removed fixed seed
6. Passed 10,000 runs

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-771](https://issues.apache.org/jira/browse/MXNET-771) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
See description

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@kalyc since `reldiff` is also used in #11685 , with this change, the test_bind() does not fail deterministically when you remove fixed seed. However, it is still flaky and will fail with `MXNET_TEST_SEED=1875642038` please take a look

**Comment by haojin2** [comment]
How many trials have you tried with your new rtol and atol values?

**Comment by roywei** [comment]
@haojin2  10,000 runs, updated in description

File: tests/python/unittest/test_executor.py [hunk 1]
```diff
@@ -118,20 +112,20 @@ def check_bind(disable_bulk_exec):
     check_bind(False)
 
 
-@with_seed(0)
+@with_seed()
```
Comment (haojin2) [inline tests/python/unittest/test_executor.py]: one nit: please also add something similar to this: https://github.com/apache/incubator-mxnet/pull/11888/files#diff-cb652780258e73a9cd08568f38929aa2R2509 to keep track of this flaky test.

**Pull request merged**